homeassistant/components/template/__init__.py

```diff
@@ -165,3 +177,32 @@ async def init_coordinator(
 
     if coordinator_tasks:
         hass.data[DATA_COORDINATORS] = await asyncio.gather(*coordinator_tasks)
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate the options from config entry data."""
+    _LOGGER.debug("Migrating from version %s:%s", entry.version, entry.minor_version)
+    data: dict[str, Any] = dict(entry.data)
+    options: dict[str, Any] = dict(entry.options)
+    if entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False
+
+    if entry.version == 1 and entry.minor_version < 2:
+        # Migrate switch value_template to state.
+        if options.get("template_type") == SWITCH_DOMAIN:
+            if options.get(CONF_VALUE_TEMPLATE) is not None:
+                options[CONF_STATE] = options.pop(CONF_VALUE_TEMPLATE)
```

Comment:
We don't need the two levels of nesting, we can just chain the conditions.
Also, the point of minor version migration is to allow downgrading. This means we should not pop the `value_template`.